api/client/pull.go

```diff
@@ -59,7 +59,7 @@ func (cli *DockerCli) CmdPull(args ...string) error {
 	authConfig := cli.resolveAuthConfig(repoInfo.Index)
 	requestPrivilege := cli.registryAuthenticationPrivilegedFunc(repoInfo.Index, "pull")
 
-	if isTrusted() && !ref.HasDigest() {
+	if trust_should_lookup(ref, repoInfo.Official) {
```

Comment:
Yeah it should... I had a change staged locally I thought I had pushed. Should be good now.